extensions/vertx-http/deployment-spi/src/main/java/io/quarkus/vertx/http/deployment/spi/RouteBuildItem.java

```diff
@@ -0,0 +1,366 @@
+package io.quarkus.vertx.http.deployment.spi;
+
+import java.util.OptionalInt;
+import java.util.function.Consumer;
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.ConfigProvider;
+
+import io.quarkus.builder.item.MultiBuildItem;
+import io.vertx.core.Handler;
+import io.vertx.ext.web.Route;
+import io.vertx.ext.web.RoutingContext;
+
+/**
+ * A build item that represents a route that should be added to the router.
+ * <p>
+ * Producing this build item does not mean the HTTP server is available.
+ * It will be consumed if the Quarkus Vert.x HTTP extension is present.
+ */
+public final class RouteBuildItem extends MultiBuildItem {
+
+    /**
+     * The type of route handler
+     */
+    public enum HandlerType {
+
+        /**
+         * A regular route handler invoked on the event loop.
+         *
+         * @see io.vertx.ext.web.Route#handler(Handler)
+         */
+        NORMAL,
+        /**
+         * A blocking route handler, invoked on a worker thread.
+         *
+         * @see io.vertx.ext.web.Route#blockingHandler(Handler)
+         */
+        BLOCKING,
+        /**
+         * A failure handler, invoked when an exception is thrown from a route handler.
+         * This is invoked on the event loop.
+         *
+         * @see io.vertx.ext.web.Route#failureHandler(Handler)
+         */
+        FAILURE
+
+    }
+
+    /**
+     * Type of routes.
+     */
+    public enum RouteType {
+
+        /**
+         * Framework routes are provided by the Quarkus framework (or extensions).
+         * They are not related to the application business logic, but provide a non-functional feature (health, metrics...).
+         * <p>
+         * Framework route can be mounted on the application router (under the non application route path) or on the management
+         * router when enabled.
+         */
+        FRAMEWORK_ROUTE,
+        /**
+         * Application routes are part of the application business logic.
+         * They are mounted on the application router (so the application prefix is applied).
+         */
+        APPLICATION_ROUTE,
+        /**
+         * Absolute routes are part of the application business logic, and are mounted on the root router (exposed on /).
+         */
+        ABSOLUTE_ROUTE
+    }
+
+    private RouteType typeOfRoute = RouteType.APPLICATION_ROUTE;
```

Comment:
Hm, I believe that build items should be immutable. In theory, the build item producer can (even mistakenly) modify the build item through the builder after it's produced and that may lead to unexpected errors.